### ISSUE TITLE
DesignForExtension: method javadoc is recognized if no modifier in on method

### ISSUE THREAD

**Issue by MihneaS** [checkstyle/checkstyle#5255]
http://checkstyle.sourceforge.net/config_design.html#DesignForExtension

```
/var/tmp $ javac BadStudent.java
/var/tmp $ cat main/BasStudent.java
public class BadStudent {
    private String name;
    public BadStudent(final String name) {
        this.name = name;
    }

    /**
     * Bad BadStudent
     */
    void putSuffix(final String sufix) {
        name = new String(name + sufix);
    }
}

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
                "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="DesignForExtension"/>
    </module>
</module>

/var/tmp $ java -jar checkstyle-8.4-all.jar -c simple_config.xml BadStudent.java

Starting audit...
[ERROR] /var/tmp/BadStudent.java:10:5: Class 'BadStudent' looks like designed for extension (can be subclassed), but the method 'putSuffix' does not have javadoc that explains how to do that safely. If class is not designed for extension consider making the class 'BadStudent' final or making the method 'putSuffix' static/final/abstract/empty, or adding allowed annotation for the method. [DesignForExtension]
Audit done.
Checkstyle ends with 1 errors.
```

---------------

I expected to have no error since the javadoc was present. With any modifer, public, protected etc, the same javadoc is recognized by checkstyle.

**Comment by romani** [comment]
@MihneaS , thanks for reporting issue, if you see where is a problem in Check, please be welcome with PR.
Might be rated to #5252 .

**Comment by pbludov** [comment]
This is by design:
```
No, they aren't related. DesignForExtensionCheck is broken by itself:

public class DesignForExtensionCheck extends AbstractCheck {
...
 /**
     * Checks whether a method has a javadoc comment.
     * @param methodDef method definition token.
     * @return true if a method has a javadoc comment.
     */
    private static boolean hasJavadocComment(DetailAST methodDef) {
        final DetailAST modifiers = methodDef.findFirstToken(TokenTypes.MODIFIERS);
        return modifiers.branchContains(TokenTypes.BLOCK_COMMENT_BEGIN);
    }
}
```
As you can see, no modifiers -> no javadoc.

**Comment by pbludov** [comment]
$ java -jar target/checkstyle-8.5-SNAPSHOT-all.jar -J /tmp/BadStudent.java 
```
    |--METHOD_DEF -> METHOD_DEF [10:4]
    |   |--MODIFIERS -> MODIFIERS [10:4]
    |   |--TYPE -> TYPE [10:4]
    |   |   |--BLOCK_COMMENT_BEGIN -> /* [7:4]
    |   |   |   |--COMMENT_CONTENT -> *\n     * Bad BadStudent\n      [7:6]
    |   |   |   |   `--JAVADOC -> JAVADOC [7:7]
    |   |   |   |       |--NEWLINE -> \n [7:7]
    |   |   |   |       |--LEADING_ASTERISK ->      * [8:0]
    |   |   |   |       |--TEXT ->  Bad BadStudent [8:6]
    |   |   |   |       |--NEWLINE -> \n [8:21]
    |   |   |   |       |--TEXT ->       [9:0]
    |   |   |   |       `--EOF -> <EOF> [9:5]
    |   |   |   `--BLOCK_COMMENT_END -> */ [9:5]
    |   |   `--LITERAL_VOID -> void [10:4]
```
after replace void putSuffix(final String sufix) =>  **public** void putSuffix(final String sufix)
```
    |--METHOD_DEF -> METHOD_DEF [10:4]
    |   |--MODIFIERS -> MODIFIERS [10:4]
    |   |   |--BLOCK_COMMENT_BEGIN -> /* [7:4]
    |   |   |   |--COMMENT_CONTENT -> *\n     * Bad BadStudent\n      [7:6]
    |   |   |   |   `--JAVADOC -> JAVADOC [7:7]
    |   |   |   |       |--NEWLINE -> \n [7:7]
    |   |   |   |       |--LEADING_ASTERISK ->      * [8:0]
    |   |   |   |       |--TEXT ->  Bad BadStudent [8:6]
    |   |   |   |       |--NEWLINE -> \n [8:21]
    |   |   |   |       |--TEXT ->       [9:0]
    |   |   |   |       `--EOF -> <EOF> [9:5]
    |   |   |   `--BLOCK_COMMENT_END -> */ [9:5]
    |   |   `--LITERAL_PUBLIC -> public [10:4]
    |   |--TYPE -> TYPE [10:11]
    |   |   `--LITERAL_VOID -> void [10:11]
```

**Comment by RoMaNiAsHkA** [comment]
Do you mean that grammar is not valid? 
Should i fix it?

**Comment by romani** [comment]
Grammar is ok, just a bug to detected where javadoc could be located.
Please be wellcome with fix.

**Comment by romani** [comment]
Fix is merged
